File: frontend/lib/career.ts
import { apiDelete, apiGet, apiPost } from "@/lib/api";

export interface CareerRequirement { skill_id:string; skill_name:string; skill_slug:string; skill_category:string; importance:string; min_mastery_required:number; min_mastery_percentage:number; target_mastery:number; target_mastery_percentage:number; relevance_note:string|null; order_index:number }
export interface CareerRole { id:string; title:string; slug:string; description:string; short_description:string; category:string; seniority_level:string; average_salary_usd:number|null; demand_level:string; typical_companies:string[]; responsibilities:string[]; related_role_slugs:string[]; is_featured:boolean; skill_requirements:CareerRequirement[] }
export interface SkillReadiness { skill_id:string; skill_name:string; skill_slug:string; category:string; importance:string; current_mastery:number; current_mastery_percentage:number; min_required:number; min_required_percentage:number; target_mastery:number; target_mastery_percentage:number; gap:number; gap_percentage:number; skill_readiness:number; status:"ready"|"close"|"gap"|"not_started"; relevance_note:string|null }
export interface CareerReadiness { role_id:string; role_title:string; role_slug:string; overall_readiness:number; overall_readiness_percentage:number; essential_readiness:number; important_readiness:number; readiness_level:string; skills:SkillReadiness[]; ready_skills:SkillReadiness[]; close_skills:SkillReadiness[]; gap_skills:SkillReadiness[]; not_started_skills:SkillReadiness[]; critical_gaps:SkillReadiness[]; estimated_weeks_to_ready:number|null }
export interface ActionPlan { role_id:string; role_title:string; executive_summary:string; estimated_job_ready_weeks:number; priority_phases:Array<{phase_number:number;title:string;duration_weeks:number;focus_skills:string[];description:string;key_actions:string[]}>; quick_wins:Array<{skill:string;skill_slug:string;current:number;required:number;gap_percentage:number;action:string;estimated_days:number}>; encouragement:string; market_note:string; generated_at:string }
export interface CareerGoalBundle { career_goal:{id:string;role_id:string;role_title:string;role_slug:string;target_date:string|null;job_ready_alert?:boolean}; role:CareerRole; readiness:CareerReadiness; readiness_change:number; roadmap_alignment?:{missing_from_roadmap:string[];suggestion:string} }
export interface MarketInsights { role_title:string; demand_description:string; key_skills_in_demand:string[]; emerging_technologies:string[]; typical_interview_topics:string[]; portfolio_recommendations:string[]; disclaimer:string }

export const getCareerRoles=(category?:string)=>apiGet<CareerRole[]>(`/api/v1/career/roles${category?`?category=${encodeURIComponent(category)}`:""}`);
export const getCareerReadiness=(id:string)=>apiGet<CareerReadiness>(`/api/v1/career/roles/${id}/readiness`);
export const getCareerActionPlan=(id:string)=>apiGet<ActionPlan>(`/api/v1/career/roles/${id}/action-plan`);
export const getCareerMarketInsights=(id:string)=>apiGet<MarketInsights>(`/api/v1/career/roles/${id}/market-insights`);
export const compareCareerRoles=(ids:string[])=>apiPost<CareerReadiness[]>("/api/v1/career/compare",{role_ids:ids});
export const setCareerGoal=(role_id:string,job_ready_alert=false,target_date?:string)=>apiPost<CareerGoalBundle>("/api/v1/career/goal",{role_id,job_ready_alert,target_date});
export const getCareerGoal=()=>apiGet<CareerGoalBundle>("/api/v1/career/goal");
export const deleteCareerGoal=()=>apiDelete<void>("/api/v1/career/goal");

File: backend/app/schemas/career.py
"""Career mapping API payloads."""
from datetime import date,datetime
from pydantic import BaseModel,Field

class CareerSkillRequirementResponse(BaseModel):
    skill_id:str;skill_name:str;skill_slug:str;skill_category:str;importance:str;min_mastery_required:float;min_mastery_percentage:int;target_mastery:float;target_mastery_percentage:int;relevance_note:str|None;order_index:int
class CareerRoleResponse(BaseModel):
    id:str;title:str;slug:str;description:str;short_description:str;category:str;seniority_level:str;average_salary_usd:int|None;demand_level:str;typical_companies:list[str];responsibilities:list[str];related_role_slugs:list[str];is_featured:bool;skill_requirements:list[CareerSkillRequirementResponse]=Field(default_factory=list)
class SkillReadinessItem(BaseModel):
    skill_id:str;skill_name:str;skill_slug:str;category:str;importance:str;current_mastery:float;current_mastery_percentage:int;min_required:float;min_required_percentage:int;target_mastery:float;target_mastery_percentage:int;gap:float;gap_percentage:int;skill_readiness:float;status:str;relevance_note:str|None
class CareerReadinessResponse(BaseModel):
    role_id:str;role_title:str;role_slug:str;overall_readiness:float;overall_readiness_percentage:int;essential_readiness:float;important_readiness:float;readiness_level:str;skills:list[SkillReadinessItem];ready_skills:list[SkillReadinessItem];close_skills:list[SkillReadinessItem];gap_skills:list[SkillReadinessItem];not_started_skills:list[SkillReadinessItem];critical_gaps:list[SkillReadinessItem];estimated_weeks_to_ready:int|None
class ActionPlanResponse(BaseModel):
    role_id:str;role_title:str;executive_summary:str;estimated_job_ready_weeks:int;priority_phases:list[dict];quick_wins:list[dict];encouragement:str;market_note:str;generated_at:datetime
class CareerGoalSetRequest(BaseModel):role_id:str;target_date:date|None=None;job_ready_alert:bool=False
class CareerCompareRequest(BaseModel):role_ids:list[str]=Field(min_length=2,max_length=3)
class UserCareerGoalResponse(BaseModel):
    id:str;role_id:str;role_title:str;role_slug:str;is_primary:bool;initial_readiness:float|None;current_readiness:float|None;target_date:date|None;created_at:datetime
class MarketInsightsResponse(BaseModel):role_title:str;demand_description:str;key_skills_in_demand:list[str];emerging_technologies:list[str];typical_interview_topics:list[str];portfolio_recommendations:list[str];disclaimer:str
